src/plugins/discover/public/application/components/sidebar/lib/get_index_pattern_field_list.ts

```diff
@@ -36,18 +36,5 @@ export function getIndexPatternFieldList(
   fieldCounts?: Record<string, number>
 ) {
   if (!indexPattern || !fieldCounts) return [];
-
-  const fieldNamesInDocs = Object.keys(fieldCounts);
-  const fieldNamesInIndexPattern = indexPattern.fields.getAll().map((fld) => fld.name);
-  const unknownTypes: IndexPatternField[] = [];
-
-  difference(fieldNamesInDocs, fieldNamesInIndexPattern).forEach((unknownFieldName) => {
-    unknownTypes.push({
-      displayName: String(unknownFieldName),
-      name: String(unknownFieldName),
-      type: 'unknown',
-    } as IndexPatternField);
-  });
-
-  return [...indexPattern.fields.getAll(), ...unknownTypes];
```

Comment:
Why did you remove this?

Reply:
fieldNamesInDocs has all the fields from prev index pattern. This difference will compare fields and add prev idx fields to unknownTypes and return them. This is why we see unknownTypes (grey fields) when we switch index pattern. I think we agree that for discover, we only need selected index pattern fields for now.

Reply:
Fair point :)